alf/utils/summary_utils.py

```diff
@@ -98,7 +101,8 @@ def histogram_continuous(name,
         bucket_min +
         (torch.arange(bucket_count + 1, dtype=torch.float64) / bucket_count) *
         (bucket_max - bucket_min))
-    data = data.clamp(bucket_min, bucket_max)
+    eps = 1e-6 if edge_inclusive else 0
+    data = data.clamp(bucket_min + eps, bucket_max - eps)
```

Comment:
Wouldn't it be better to add eps to the bucket min and max instead? Though since it's such a small number, shouldn't matter either way.

Reply:
> Wouldn't it be better to add eps to the bucket min and max instead? Though since it's such a small number, shouldn't matter either way.

I think they are equivalent.

Reply:
I was thinking this would be an issue if mean values are also repeated by mousing over the histogram bars, but we report mean separately anyways so should be fine.

Reply:
I also feel changing the boundary slightly is better. The change will be simpler and the resulted histogram is more likely to be same as the desired one.

Reply:
> I also feel changing the boundary slightly is better. The change will be simpler and the resulted histogram is more likely to be same as the desired one.

Now updated to extend the leftmost/rightmost edge.